tests/worker/test_reworker.py

```diff
@@ -0,0 +1,127 @@
+import itertools
+from concurrent.futures import Future
+from typing import Callable, Iterable, List, Optional, TypeVar
+
+import pytest
+
+from blueapi.core import BlueskyContext, EventStream
+from blueapi.worker import (
+    RunEngineWorker,
+    RunPlan,
+    Task,
+    TaskStatus,
+    Worker,
+    WorkerEvent,
+    WorkerState,
+)
+
+
+@pytest.fixture
+def context() -> BlueskyContext:
+    ctx = BlueskyContext()
+    ctx.with_startup_script("blueapi.startup.example")
+    return ctx
+
+
+@pytest.fixture
+def worker(context: BlueskyContext) -> Iterable[Worker[Task]]:
+    worker = RunEngineWorker(context)
+    yield worker
+    worker.stop()
+
+
+def test_stop_doesnt_hang(worker: Worker) -> None:
+    worker.start()
+
+
+def test_stop_is_idempotent(worker: Worker) -> None:
```

Comment:
nit-pick: I feel like if I have to google what idempotent means, and then still don't understand why this test name makes sense, maybe it's a bad test name?

Idempotent; denoting an element of a set which is unchanged in value when multiplied or otherwise operated on by itself.

Reply:
aha - ok, I get it now. You don't call worker.start(). So you're testing that worker.stop() does nothing if the worker isn't started.

So perhaps, instead, 
```python
test_stop_is_idempontent_if_worker_not_started(worker: Worker) -> None:
```